packages/core/helper-plugin/lib/src/components/RelationsInput/components/ShadowList.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { Box } from '@strapi/design-system/Box';
+import { Stack } from '@strapi/design-system/Stack';
+
+const ShadowBox = styled(Box)`
+  position: relative;
+
+  &:before {
+    content: ${({ overflowing }) =>
+      overflowing === 'both' || overflowing === 'top' ? "''" : undefined};
+    position: absolute;
+    background: linear-gradient(#dfdfe7 0%, rgba(0, 0, 0, 0) 100%);
+    opacity: 0.8;
+    top: 0;
+    width: 100%;
+    height: 4px;
+    z-index: 1;
+  }
+
+  &:after {
+    content: ${({ overflowing }) =>
+      overflowing === 'both' || overflowing === 'bottom' ? "''" : undefined};
+    position: absolute;
+    background: linear-gradient(rgba(0, 0, 0, 0) 0%, #dfdfe7 100%);
+    opacity: 0.8;
+    bottom: 0;
+    width: 100%;
+    height: 4px;
+    z-index: 1;
+  }
+`;
+
+export const ShadowList = ({ overflowing, children, ...props }) => {
+  return (
+    <ShadowBox overflowing={overflowing} {...props}>
```

Comment:
Why do you need to do prop drilling for `overflowing`? Can't we just pass it as part of `...props`?

Reply:
Mainly to have this handy proptype documentation / warning to help when using the component:
`overflow: PropTypes.oneOf(['top-bottom', 'bottom', 'top'])`

wdyt?

Reply:
I think you can still add the prop-types in that case, no?